linter.py

```diff
@@ -464,9 +487,25 @@ def linter_check(repo_root, linter_subfolder):
 
         if not (cpp_lint_success and pylint_success):
             print("=" * 80)
-            print("Commit rejected! Please address the linter errors above.")
+            print("Commit not up to standards!")
+            print("Please address the linter errors above.")
             print("=" * 80)
-            exit(1)
+            if linter_config['block-commits']:
```

Comment:
does this also trigger blocking behavior if the config does not contain this setting?

Reply:
Yes, because the function `read_linter_config` sets `config = DEFAULT_CONFIG` which contains `'block-commits': True`. This is afaik also the behavior we had before our change.